iroh/tests/spaces.rs

```diff
@@ -132,3 +136,87 @@ async fn spaces_smoke() -> Result<()> {
 
     Ok(())
 }
+
```

Comment:
This test fails on this branch, but works in the `willow` branch.
It's waiting on `syncs.complete_all().await`, which never completes (neither with `SessionMode::ReconcileOnce` nor `SessionMode::Continuous`).
The test succeeds if we share the ticket using `RestrictArea::None` instead of `RestrictArea::Restrict(...)`.
```suggestion
#[tokio::test]
async fn test_regression_restricted_area_sync() -> testresult::TestResult {
    iroh_test::logging::setup_multithreaded();

    const TIMEOUT: std::time::Duration = std::time::Duration::from_secs(2);

    let (alfie_addr, alfie) = spawn_node().await;
    let (betty_addr, betty) = spawn_node().await;
    info!("alfie is {}", alfie_addr.node_id.fmt_short());
    info!("betty is {}", betty_addr.node_id.fmt_short());

    let alfie_user = alfie.spaces().create_user().await?;
    let betty_user = betty.spaces().create_user().await?;

    let alfie_space = alfie
        .spaces()
        .create(NamespaceKind::Owned, alfie_user)
        .await?;

    let space_ticket = alfie_space
        .share(
            betty_user,
            AccessMode::Write,
            // RestrictArea::None, // succeeds with this
            RestrictArea::Restrict(Area::new_subspace(betty_user)),
        )
        .await?;

    let (betty_space, syncs) = betty
        .spaces()
        .import_and_sync(space_ticket, SessionMode::ReconcileOnce)
        .await?;

    let completion = tokio::time::timeout(TIMEOUT, syncs.complete_all()).await?;
    println!("Completed syncs: {completion:#?}");

    let stream = betty_space.get_many(Range3d::new_full()).await?;
    let entries: Vec<_> = stream.try_collect().await?;
    println!("{entries:#?}");

    Ok(())
}

```

Reply:
I've also seen `Space::sync_once` just hang when two nodes try to sync that have never exchanged capabilities.

Reply:
I think the test failed to pass because this branch was missing a commit from main: #2683 
Or at least the test passes just fine for me after rebasing this branch on willow!

Reply:
> I've also seen Space::sync_once just hang when two nodes try to sync that have never exchanged capabilities.

I think this is currently expected behavior: PAI waits forever for matching capabilities to be submitted. We need to close our side after we have submitted everything I think. Aljoscha made some recent modifications to WGPS to support this properly (I still have to check them out)

Reply:
The test passes reliably for me.